core/src/main/java/org/ehcache/spi/cache/Store.java

```diff
@@ -441,7 +441,7 @@
      * Informs this Provider, a Store it created is being disposed (i.e. closed)
      * @param resource
      */
-    void releaseStore(Store<?, ?> resource);
+    void releaseStore(Store<?, ?> resource) throws CacheAccessException;
```

Comment:
Anything thrown here is going to propagate all the the way out the the user wrapped in a `StateTransitionException`.  I'm not averse to having these methods throw when bad things happen, but to me two things need to happen.  Firstly we can't be throwing eagerly, and secondly I don't think we should be throwing CacheAccessException... it's an internal exception that we shouldn't be leaking (even if it is wrapped).  Does this sound sensible?